test/fortran/CMakeLists.txt

```diff
@@ -1,25 +1,15 @@
 add_library(test_utils STATIC utils.f90)
-
-# Generated ctest file that switches and displays tests
-# Note must be plain C because generated CPP file does not include extern "C"
-set(TestsDriver fortran_test.c)
-
-# Add fortran tests here
-create_test_sourcelist(FortranTests_Files
-		${TestsDriver}
-		test_fortran_spg_get_symmetry.f90)
```

Comment:
I would prefer to either use `create_test_sourcelist` or use an upstream Fortran unit tester like Fortuno.

This would minimize the executables displayed in IDEs, faster to build, and it is easier to add tests (less poilerplate compared to writing `Program` for each one)

Reply:
I see your point.

I could not figure out how to put multiple test functions in one file, which is for easier writing and reading the test code. Using `create_test_sourcelist`, it registers one function per one file, as far as I tried. So if I want to add more tests, I thought I had to create a number of files. If this point is overcome, surely what you explained is better.

Reply:
Totally agree with you. I would like it in a similar file as well.

Here is a design choice to consider though. Do we want `ctest` to be able to call subfunctions as well?

For example:
```cmake
# Loop over all main tests
foreach (test ${TestsToRun})
  get_filename_component(test_name ${test} NAME_WE)
  # Check if we have subtests defined
  if (${SubTests_${test_name}})
    # Loop over all subtests
    foreach (subtest ${SubTests_${test_name}}
      # Call test via ${FortranTests_Executable} ${test_name} ${subtest}
      # E.g.: ./FortranTests_Executable test_fortran_spg_get_symmetry test_rutile112
      add_test(NAME "${test_name}/${subtest}"
               COMMAND $<TARGET_FILE:FortranTests> ${test_name} ${subtest})
    endforeach ()
  else ()
    # Only run main test
    add_test(NAME "${test_name}"
             COMMAND $<TARGET_FILE:FortranTests> ${test_name})
  endif ()
endforeach ()
```

I can show you how to implement that from the `argc`/`argv` if you need help on that.

Reply:
I see. It is a good trick. If I understand it correctly, I need two lists, `${TestsToRun}` and `${SubTests_${test_name}`. Am I correct?

> I can show you how to implement that from the argc/argv if you need help on that.

Yes, I need your help. I would appreciate it.

Reply:
Ok, I'll push a commit directly here with an example for `test_fortran_spg_get_symmetry` maybe tonight or tomorrow.